enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,330 @@
+within an Openshift pod without being in a privileged namespace.
+
+### User Stories
+
+* As an Openshift user, I would like to be able to run my container as root without needing to be trusted by the platform.
+* As a user of Openshift Devspaces, I would like to run podman within the Devspace.
+* As an Openshift admin, I would like to run untrusted users on a tighter security profile than the SCC restricted-v2.
+* As an Openshift admin, I would like to ensure pods that request user namespaces are confined to one.
+
+### Goals
+
+- Enable Openshift users to request a pod be put in a user namespace
+- Update SCC to take user namespaces into account when choosing the security profile of a container
+- Add support for users to run podman within an Openshift pod without being privileged.
+
+### Non-Goals
+
+- 
+
+## Proposal
+
+There are two pieces to this proposal:
+- Extend SCC to be aware of the hostUsers field:
+    - Add a new field HostUsers to SCC, which will be Allowed by default
+    - Add a new SCC to the default list: restricted-v3
+        - This SCC will be identical to restricted-v2, but have HostUsers set to Disallowed
+    - Add a new SCC to the default list: nonroot-v3
+        - This SCC will be identical to nonroot-v2, but have HostUsers set to Disallowed
+    - Add a new SCC to the default list: nested-container. It will have:
+        - SELinux context set to MustRunAs.Type: container_engine_t
+        - RunAsUserStrategy: RunAsAny
+        - RequiredDropCapabilities: None
+        - HostUsers: Disallowed
+        - And otherwise mirror the restricted-v2 profile
+- Add the features UserNamespacesPodSecurityStandards and ProcMountType to the list that qualify a cluster as TechPreviewNoUpgrade
+    - UserNamespacesSupport will be on by default, as the feature is beta in 1.30.
+
+### Workflow Description
+
+N/A
+
+### API Extensions
+
+- Add HostUsers to SCC. This relies on approval from the apiserver team.
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### SecurityContextConstraint Updates
+
+##### `HostUsers` field
+
+Add the following to the [Openshift API](https://github.com/openshift/api/blob/76a71dac36a08eab1b240c6c8d4e39c813b1b12b/security/v1/types.go):
+
+```
+       // hostUsers determines if the policy allows host users in containers.
+       // Valid values are "Allowed", "Disallowed" and omitted.
+       // When omitted, this means no opinion and the platform is left to choose a reasonable default.
+       // The current default is "Allowed".
+       // +openshift:enable:FeatureGate=UserNamespacesPodSecurityPolicy
+       // +kubebuilder:validation:Enum="Allowed";"Disallowed";""
+       // +kubebuilder:validation:Optional
```

Comment:
Confusingly, we use `+optional` instead of this version, as it has wider support, and `+kubebuilder:validation:Required` for the required tag... I'm working to solve that

Reply:
updated